utils/global-state-update-gen/src/generic.rs

```diff
@@ -177,8 +177,12 @@ fn gen_snapshot_from_old(
             Some(validator) => {
                 *recipient = SeigniorageRecipient::new(
                     validator.bonded_amount,
-                    validator.delegation_rate,
-                    validator.delegators_map(),
+                    validator
+                        .delegation_rate
+                        .unwrap_or(*recipient.delegation_rate()),
```

Comment:
I think this code would be clearer if instead of `snapshot` the variable name made it clear it's the _old snapshost_ and the `stakes_map` might overwrite it. Just by looking at these few lines it isn't exactly easy to understand why we're using `recipient.delegation_rate()` if `validator.delegation_rate` is `None`.